### DISCUSSION TITLE
lab4-2 fir.c與user_proj_wrapper內容

### DISCUSSION THREAD

**Discussion by BarryJu1999** [bol-edu/HLS-SOC-Discussions#137] [Q&A]
在討論區有先看到此篇文章，底下助教留言有寫到要將#define的指令放在counter_la_fir.c中，
https://github.com/bol-edu/HLS-SOC-Discussions/discussions/127

但似乎與老師在影片中後半部提到的不太一樣 https://www.youtube.com/watch?v=f9zD3Fs9Bhk&t=2252s (後半部提到的不太一樣)，
這樣意思在收發X和Y時(或其他韌體相關)都寫在counter_la_fir.c然後fir.c和fir.h就寫的跟4-1的一樣？
另外不太理解助教提到user_proj_wrapper包起來是指直接把lab3的code放進去接線還是應該怎麼做？
謝謝大家。

**Comment by LingChiYang** [comment]
同學你好

很抱歉周末的檔案更新，和老師討論過後，我們希望同學自行新增fir_control.c寫4-2的firmware。先前若有誤會的同學，我想也影響不大，只是firmware換個地方寫。可參考
https://github.com/bol-edu/caravel-soc_fpga-lab/blob/main/lab-caravel_fir/lab4-2-caravel-testbench.pdf

user_proj_wrapper如同workbook的內容需要包含4-1和4-2的接口，Lab3需要改一些內容再放進來，protocol是不變的。

**Reply by BarryJu1999** [reply]
那所以fir.c需要寫任何東西嗎，還是是將lab4-1的內容貼上就好？
另外fir_control.是放在原fir.c同一個目錄還是放在firmware的資料夾中呢？
這樣run_sim是否需要修改？
謝謝。

**Reply by LingChiYang** [reply]
對fir.c fir.h是4-1的。fir_control.c放在同目錄。至於run_sim需不需要修改，同學跑跑看就知道答案了

**Reply by Lanxas1018** [reply]
同學好，4-1的fir.c和4-2所需要用的firmware code沒有關係，在4-1中我們是用軟體的方式去刻fir filter，但到了4-2我們會將lab3中所設計的RTL fir filter放進來，所以原先4-1 fir.c的code完全不會使用到，4-2中我們並沒有要求同學的firmware code一定要寫在哪份檔案中，但可以根據提供的文件來做參考，再次提醒4-2要寫的firmware是根據spec控制整個control flow，而非fir filter design
